versions/development/SPEC.md

```diff
@@ -1218,81 +1257,343 @@ $runtime = 'runtime' $ws* '{' ($ws* $runtime_kv $ws*)* '}'
 $runtime_kv = $identifier $ws* '=' $ws* $expression
 ```
 
-The runtime section defines key/value pairs for runtime information needed for this task.  Individual backends will define which keys they will inspect so a key/value pair may or may not actually be honored depending on how the task is run.
 
-Values can be any expression and it is up to the engine to reject keys and/or values that do not make sense in that context.  For example, consider the following WDL:
+The runtime section defines a set of key/value pairs which represent the minimum requirements needed to run a task or, which define the conditions under which a task should be interpreted as a failure or success. All keys within the runtime section are well defined, and must be honored by the execution engine. This also means that arbitrary key/value pairs within the runtime section are not allowed, and should be rejected by an engine. Arbitrary Key/value pairs should be added to the `hints` section instead.
+
+During execution of a task, resource requirements within the runtime section must be enforced by the engine. If the engine is not able to provision the requested resources, then the task should immediately fail prior to the start of execution. In order to ensure adequate information is provided to execute a task, there is one required key, and strict default values for optional keys, which are defined below. Default values for all optional keys are directly defined by the WDL specification in order to encourage portability of workflows and tasks. Execution engines should NOT provide additional mechanisms to set _default_ values for when no runtime attributes are defined.
+
+Values can be any expression which evaluate to a valid entry for that runtime property. Since values are expressions, they can also reference variables in the task:
 
 ```wdl
 task test {
+  input {
+    String ubuntu_version
+  }
   command {
     python script.py
   }
+  
   runtime {
-    docker: ["ubuntu:latest", "broadinstitute/scala-baseimage"]
+    container: ubuntu_version
   }
 }
 ```
 
-The value for the `docker` runtime attribute in this case is an array of values.  The parser should accept this.  Some engines might interpret it as an "either this image or that image" or could reject it outright.
 
-Since values are expressions, they can also reference variables in the task:
+#### container (**Required**)
+
+The container key must accept one ore more locations which inform the engine where to retrieve a container image to execute the task. It is expected that all container images provided are identical. They MUST produce the same final results when the task is run. It is the responsibility of the individual execution engine to define the specific image sources which it supports, and to determine which image is the "best" one to use at runtime. Defining multiple images enables greater portability across a broad range of execution environments. 
```

Comment:
Similarly to above, I'm unclear on the need for capitalization (or more precisely, inconsistent capitalization)